### PR TITLE
Implement secant distribution

### PR DESCRIPTION
**Pull request by mmikhasenko**
Implement the Hyperbolic Secant distribution as a new continuous univariate distribution.

---
<a href="https://cursor.com/background-agent?bcId=bc-3a1afdb5-0fa3-42bc-8c3a-163f7627448a">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">
    Open in Cursor
  </picture>
</a>
<a href="https://cursor.com/agents?id=bc-3a1afdb5-0fa3-42bc-8c3a-163f7627448a">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">
    Open in Web
  </picture>
</a>

<sub>[Learn more](https://docs.cursor.com/background-agent/web-and-mobile) about Cursor Agents</sub>

### CODE REVIEW

**Review by mmikhasenko — COMMENTED** [review]
The implementation looks good. In addition to minor comments there is one bug, the Kurtosis in Distributions.jl really means the excess kurtosis. Implementation and the tests should be updated

File: test/test-secant.jl [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        # Test inflection points (approximately at μ ± 0.561σ)
+        # At inflection points, second derivative should be zero
+        inflection_distance = 2 / π * log(√2 + 1)
+        @test inflection_distance ≈ 0.5615 atol = 1e-3
+    end
+
+    @testset "Edge Cases" begin
+        # Test behavior near boundaries of quantile function
+        @test_throws ArgumentError quantile(d_standard, 0.0)
+        @test_throws ArgumentError quantile(d_standard, 1.0)
```
[the remark this answers is not in the document]
Comment (mmikhasenko) [reply test/test-secant.jl]: updated further

`==0` => `-Infty`,
`==1` => `+Infty`,

error away from [0,1]

**Comment by mmikhasenko** [comment]
@peremato 

one more pair of eyes (yours) is requested.

- the basic code is written by a language model in cursor based on #29 
- I've reviewed the code locally, and also did a code review in GitHub, fixing several issues.

I think it's good to go.
